Source/Core/BoundingSphere.js

```diff
@@ -1198,6 +1198,33 @@ define([
         return result;
     };
 
+    /**
+     * @private
+     */
+    BoundingSphere.closestLocationIn = function(position, boundingObject) {
+        if (defined(boundingObject)) {
+            var center = boundingObject.center;
+            var radius = boundingObject.radius;
+
+            // to avoid dividing by zero check if same location as center
+            if (position.equals(center)) {
+                return position;
+            }
+            // check if already within sphere
+            if (Cartesian3.distance(position, center) < radius) {
+                return position;
+            }
+
+            var direction = new Cartesian3();
+            Cartesian3.subtract(position, center, direction);
+            Cartesian3.normalize(direction, direction);
+            Cartesian3.multiplyByScalar(direction, radius, direction);
+
+            position = Cartesian3.add(direction, center, new Cartesian3());
```

Comment:
Have the functions take a `result` parameter so we're not creating a `new Cartesian3()`